server/remote_cache/cachetools/cachetools_test.go

```diff
@@ -555,3 +562,152 @@ func (b *bsReadStreamer) SendMsg(m any) error {
 func (b *bsReadStreamer) Trailer() metadata.MD {
 	panic("unimplemented")
 }
+
+func TestUploadReaderAndGetBlob(t *testing.T) {
+	randomStringLargerThan1MB, err := random.RandomString(2 * 1024 * 1024)
+	require.NoError(t, err)
+	for _, tc := range []struct {
+		name              string
+		input             string
+		useZstd           bool
+		expectUploadError bool
+		uploadSize        int64
+		expectGetError    bool
+		getSize           int64
+		expectedGetSize   int64
+	}{
+		{
+			name:              "simple upload and get",
+			input:             "simple upload and get",
+			useZstd:           false,
+			expectUploadError: false,
+			uploadSize:        int64(len([]byte("simple upload and get"))),
+			expectGetError:    false,
+			getSize:           int64(len([]byte("simple upload and get"))),
+			expectedGetSize:   int64(len([]byte("simple upload and get"))),
+		},
+		{
+			name:              "upload with incorrect size fails",
+			input:             "upload with incorrect size fails",
+			useZstd:           false,
+			expectUploadError: true,
+			uploadSize:        int64(len([]byte("upload with incorrect size fails"))) - 4,
+			expectGetError:    true,
+			getSize:           int64(len([]byte("upload with incorrect size fails"))),
+			expectedGetSize:   int64(len([]byte("upload with incorrect size fails"))),
+		},
+		{
+			name:              "get with incorrect size still succeeds",
+			input:             "get with incorrect size still succeeds",
+			useZstd:           false,
+			expectUploadError: false,
+			uploadSize:        int64(len([]byte("get with incorrect size still succeeds"))),
+			expectGetError:    false,
+			getSize:           int64(len([]byte("get with incorrect size still succeeds"))) - 4,
+			expectedGetSize:   int64(len([]byte("get with incorrect size still succeeds"))),
+		},
+		{
+			name:              "zstd simple upload and get",
+			input:             "zstd simple upload and get",
+			useZstd:           false,
```

Comment:
Should be `true`? Applies to the other zstd tests too

Reply:
🤦  Thank you!